### ISSUE TITLE
Preference Classes Do Not Impact Agent Inclination to Follow Particular Agent Types

### ISSUE THREAD

**Issue by nbuhagiar** [hashkat/hashkat#30]
According to section 2.5 of the 'Agent-based simulation of complex social networks, a kinetic Monte Carlo approach' paper, preference classes determine "what proportion connections are made towards the defined agent types, and the rebroadcast probability of a given broadcast. For example, one could define a preference class to model a user on Twitter who only likes celebrities, and certain topics." However, as far as I know, preference classes in #k@ only deal with tweet transmission for particular tweet types and following via retweets, not an agent's inclination to follow a particular agent type. Is there something I'm missing, or is this something that needs to be implemented?

**Comment by ludamad** [comment]
Good point. The paper described things a bit broadly, to the point that some of the 'things you could define' require source modification. This should be implemented.

**Comment by itamblyn** [comment]
Hrm. I thought this was already present in the code. Yah, we actually need this…

**Comment by ludamad** [comment]
This would probably best be a separate agent preference. Issue #113 probably has the right idea in making the current preference class entity variable strictly about retweeting. So we'd have retweet_preference and agent_preference, instead.
